detector/muonSystem/muonSystemMuRWELL_o1_v01.cpp

```diff
@@ -0,0 +1,941 @@
+
+  double endcapDetectorYLength = endcapDetectorLayerOuterRadius - endcapDetectorLayerInnerRadius;
+  double endcapYLength = endcapRadiatorLayerOuterRadius - endcapRadiatorLayerInnerRadius; // It is the distance betwwen the inner and the outer radius of the endcap, it can be in both Y and X dimensions //it is also the same for the endcap detector layers.
+
+  //  int numCopiesY = sideLength / (2 * dimensions.y() - overlapY) + 1;
+  //  int numCopiesZ = barrelLength / (2 * dimensions.z() - overlapZ) + 1;
+
+  //  double remainderY = std::fmod((sideLength - 2 * clearance), (2 * dimensions.y() - overlapY)) / (2 * dimensions.y());
+  double remainderZ = std::fmod((barrelLength - 2 * clearance), (2 * dimensions.z() - overlapZ)) / (2 * dimensions.z()) - (2 * clearance / dimensions.z());
+  double endcapRemainderZ = std::fmod((endcapDetectorYLength - 2 * clearance), (2 * dimensions.z() - overlapZ)) / (2 * dimensions.z()) - (2 * clearance / dimensions.z());
+  //  double remainderY2 = std::fmod((sideLength2 - 2 * clearance), (2 * dimensions.y() - overlapY)) / (2 * dimensions.y());
+
+  //  double remainderYLength = remainderY * 2 * dimensions.y();
+  //  double remainderZLength = remainderZ * 2 * dimensions.z();
+
+  //  double totalOverlapValueY = (numCopiesY - 1) * overlapY;
+  //  double totalOverlapValueZ = (numCopiesZ - 1) * overlapZ;
+
+  double sideEnvX = detectorVolumeThickness/2.0; // 
+  double sideEnvY = (sideLength / 2.0); // + clearance * 15;
+  double sideEnvZ = (barrelLength / 2.0); // + clearance/2.0;
+  double sideEnvY2 = (sideLength2 / 2.0);
+
+  //  double barrelRadiatorEnvX = barrelRadiatorThickness/2.0;
+  //  double barrelRadiatorEnvY = barrelRadiatorSideLength2 / 2.0;
+  //  double barrelRadiatorEnvZ = barrelLength / 2.0;
+
+  //  double endcapRadiatorEnvX = endcapRadiatorLayerOuterRadius;  // it depends on outer radius
+  //  double endcapRadiatorEnvY = endcapRadiatorLayerOuterRadius;  // outer radius too
+  //  double endcapRadiatorEnvZ = endcapRadiatorThickness / 2.0;  // layer thickness
+
+  //  double endcapDetectorEnvX = endcapDetectorLayerOuterRadius;  // it depends on outer radius
+  //  double endcapDetectorEnvY = endcapDetectorLayerOuterRadius;  // outer radius too
+  double endcapDetectorEnvZ = detectorVolumeThickness;  // layer thickness
+
+  //  double sideY = (numCopiesY -1) * 2 * dimensions.y() + remainderYLength - totalOverlapValueY;
+  //  double sideZ = (numCopiesZ -1) * 2 * dimensions.z() + remainderZLength - totalOverlapValueZ;
+
+  //  int nameCounter = 0;
+  int barrelIdCounter = 1;
+  int endcapIdCounter = 1;
+
+// ----------------------------------------------------------------------------------------------------
+//                               // B A R R E L //
+// ----------------------------------------------------------------------------------------------------
+
+  if (numBarrelDetectorLayers > 0){
```

Comment:
Maybe it makes sense to split this into a barrel and an endcap constructor?

Reply:
Since the barrel last layer is enclosing all the endcap layers inside, we preferred to keep them in the same builder. Both Barrel and Endcap are sharing some common variables like `numSides` of the main polygon shape. and the length of the last layer of the barrel depending on the total thickness of the endcap detector. And to distinguish between the collection hits of barrel and endcap, a bit-field component has been defined. Maybe we might tend to separate them in the future if needed.